libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -653,50 +654,40 @@ internal object MapboxRouteLineUtils {
      * are cached for performance reasons.
      */
     internal val extractRouteData: (
-        route: DirectionsRoute,
+        route: NavigationRoute,
         trafficCongestionProvider: (RouteLeg) -> List<String>?
     ) -> List<ExtractedRouteData> =
-        { route: DirectionsRoute, trafficCongestionProvider: (RouteLeg) -> List<String>? ->
-            var runningDistance = 0.0
+        { route: NavigationRoute, trafficCongestionProvider: (RouteLeg) -> List<String>? ->
             val itemsToReturn = mutableListOf<ExtractedRouteData>()
-            route.legs()?.forEachIndexed { legIndex, leg ->
+            val granularDistances = granularDistancesProvider(route)!!
```

Comment:
The !! seems unwise. It doesn't cost anything to wrap everything in an ifNonNull or similar safety net.

Reply:
Good catch, leftover from a prototype :+1: